polyfills/systemxhr/systemxhr.js

```diff
@@ -192,21 +193,61 @@
       Object.defineProperty(this, property, {
         enumerable: true,
         get: function() {
+          if (property === 'response') {
+            self._convertToResponseType();
+          }
           return this['_' + property];
         },
         set: function(value) {
           this['_' + property] = value;
-          navConnHelper.methodCall({
+          if (property !== 'responseType') {
+            navConnHelper.methodCall({
                                     methodName: 'set',
                                     numParams: 2,
                                     returnValue: VoidRequest,
                                     promise: _systemxhr,
                                     field: 'xhrId'
                                   }, property, value);
+          }
         }
       });
     });
 
+    this._convertToResponseType = function() {
+      if (typeof this._response !== 'string') {
+        return;
```

Comment:
Also if what we're getting always is the responseText, why don't we store it as this._responseText?